### ISSUE TITLE
What are the minimum requirements?

### ISSUE THREAD

**Issue by gerardp** [CarterLi/iina#31]
iina works on 10.11+ but iina-plus does not work on my mac with 10.14.6.

**Comment by CarterLi** [comment]
Should be 10.12+

**Comment by CarterLi** [comment]
Found the problem is

```shell
$ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.14.6
BuildVersion:	18G9323

$ /Applications/IINA.app/Contents/MacOS/IINA
dyld: Symbol not found: ___NSDictionary0__struct
  Referenced from: /Applications/IINA.app/Contents/MacOS/../Frameworks/libavdevice.59.dylib (which was built for Mac OS X 11.0)
  Expected in: /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation
 in /Applications/IINA.app/Contents/MacOS/../Frameworks/libavdevice.59.dylib
[1]    2021 abort      /Applications/IINA.app/Contents/MacOS/IINA
```

We were using homebrew's prebuild version, however they were built for the os version that homebrew were running ( 11.0 ). That was not good...

@low-batt Have any ideas?

**Comment by low-batt** [comment]
I know what the problem is.

First, background on the issue. What we are fighting against is that a number of open source packages, especially GNU libraries, expect you to build from sources on the machine that the resulting library or executable will be run on. Homebrew takes this same attitude.

IINA's official position is that it supports macOS 10.11+. I investigated a user report of trouble on 10.11. In that case it was a GNU library whose build configuration was conditionalized for a time related method that Apple didn't add until 10.12. Since the build was done under a newer macOS version it expected that method to be present. My thoughts on this particular issue is that IINA should give up on supporting 10.11.

Homebrew will build for the macOS version it is running on. Since GitHub updated their build servers the build is now using 11.0. The Homebrew project views Homebrew as a package manager, not a build system. In particular they are strong on rejecting requests to support cross compilation and building universal binaries. But it can be forced to build for older version of macOS than the version Homebrew is running under.

How IINA does this can be seen in [compile.rb](https://github.com/iina/homebrew-mpv-iina/blob/master/other/compile.rb) in the `homebrew-mpv-iina` repository. It involves patching Homebrew code and setting `MACOSX_DEPLOYMENT_TARGET`. As it is patching Homebrew it must also take steps to prevent Homebrew from upgrading and erasing the patches.

My conclusion is doable, but painful and of course fragile as updates to Homebrew can break the patches.

I'm currently really busy looking into a hang that might be due to a mpv issue. I think I have fixed the hang during quit. Waiting to hear confirmation on that.
